
# Publish all joint states -----------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50  

# Position Controllers ---------------------------------------
# values based on dynamic tuning in RQT:
# http://gazebosim.org/tutorials/?tut=ros_control
joint_base:
  type: effort_controllers/JointPositionController
  joint: joint_base
  pid: {
    p: 100.0, 
    i: 0.01, 
    d: 10.0, 
    i_clamp: 50.0, 
    antiwindup: true
  }
  
joint0:
  type: effort_controllers/JointTrajectoryController
  joints: 
    - joint0
  gains:
    joint0: {
      p: 800.0,
      i: 500.0,
      d: 100.0,
      i_clamp: 50.0,
      antiwindup: true
    }
  
joint1:
  type: effort_controllers/JointPositionController
  joint: joint1
  pid: {
    p: 200.0,
    i: 40.0,
    d: 30.0,
    i_clamp: 50.0,
    antiwindup: true
  }
  
joint2:
  type: effort_controllers/JointPositionController
  joint: joint2
  pid: {
    p: 20.0,
    i: 10.0,
    d: 1.0,
    i_clamp: 10.0,
    antiwindup: true
  }
  
joint3:
  type: effort_controllers/JointPositionController
  joint: joint3
  pid: {
    p: 50.0,
    i: 10.0,
    d: 5.0,
    i_clamp: 100.0,
    antiwindup: true
  }
  
joint4:
  type: effort_controllers/JointPositionController
  joint: joint4
  pid: {
    p: 50.0,
    i: 5.0,
    d: 0.5,
    i_clamp: 10.0,
    antiwindup: true
  }
  
joint5:
  type: effort_controllers/JointPositionController
  joint: joint5
  pid: {
    p: 50.0,
    i: 10.0,
    d: 0.5,
    i_clamp: 10.0,
    antiwindup: true
  }
  
joint6:
  type: effort_controllers/JointPositionController
  joint: joint6
  pid: {
    p: 50.0,
    i: 10.0,
    d: 0.5,
    i_clamp: 10.0,
    antiwindup: true
  }
  
